server/src/main/java/org/opensearch/transport/nativeprotocol/NativeOutboundHandler.java

```diff
@@ -177,15 +177,17 @@ public void sendErrorResponse(
             false
         );
         ActionListener<Void> listener = ActionListener.wrap(() -> messageListener.onResponseSent(requestId, action, error));
-        sendMessage(channel, message, listener);
+        sendMessage(requestId, channel, message, listener);
     }
 
-    private void sendMessage(TcpChannel channel, NativeOutboundMessage networkMessage, ActionListener<Void> listener) throws IOException {
+    private void sendMessage(long reqId, TcpChannel channel, NativeOutboundMessage networkMessage, ActionListener<Void> listener)
```

Comment:
What's the purpose of the change to pass the request ID?

Nitpick, but `requestId` seems to be used everywhere else instead of `reqId`.